lib/pbench/server/api/resources/query_apis/dataset.py

```diff
@@ -100,26 +116,13 @@ def generate_actions(
             A generator for Elasticsearch bulk update actions
         """
         action = context["attributes"].action
-
-        sync = Sync(
-            logger=current_app.logger, component=context["attributes"].operation_name
-        )
-        sync.update(dataset=dataset, state=OperationState.WORKING)
-        context["sync"] = sync
         es_doc = {}
 
         if action == "update":
-            access = params.query.get("access")
-            owner = params.query.get("owner")
-            if not access and not owner:
-                raise MissingParameters(["access", "owner"])
-            if access:
-                context["access"] = es_doc["access"] = access
-            if owner:
-                authorized_user = Auth.token_auth.current_user()
-                if not authorized_user.is_admin():
-                    raise UnauthorizedAdminAccess(authorized_user, OperationCode.UPDATE)
-                context["owner"] = es_doc["owner"] = owner
+            if "access" in context:
+                es_doc["access"] = context["access"]
+            if "owner" in context:
+                es_doc["owner"] = context["owner"]
```

Comment:
While the set notation is cute and compact, this kinda feels like just "for the sake of being tricky"; and, technically, if I were to let myself be tempted, I think the `fields` local is unnecessary indirection ...

Reply:
> this kinda feels like just "for the sake of being tricky"

It is indeed very close to that (principally because there are only two fields; if there were six, it would be a rather different story...).

However, the definition of `fields` (which could be given a substantially more descriptive name...) could be placed in some common or global location, such that adding another field could be done centrally and wouldn't require adding another `if` statement here just to copy it...the utility of which prompted me to mention it.  (And, I can imagine other uses and code-generalizations based on it....)

For just two fields, it is likely overkill...the question is, if we were to add another field, would we then switch to make the code data-driven?  What about the field after that??  (I worked on a project where they had code like this, and there were about 25 or 30 fields...and, because they had added them one or a few at a time, they never got around to changing the cascading `if` statements into a table....)